bool ready_to_log